src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -197,11 +197,11 @@ public Optional<Age> getAge() {
         }
 
         public void setDetail(Detail detail) {
-            this.detail = detail;
+            this.detail = (detail != null) ? detail : null;
         }
 
         public Optional<Detail> getDetail() {
-            return Optional.ofNullable(detail);
+            return (detail != null) ? Optional.ofNullable(detail) : Optional.empty();
```

Comment:
Do you think this is necessary?
From what I understand, `Optional.ofNullable` already returns `Optional.empty()` when detail is empty.

Reply:
Thx for the suggestion! I'll remove the redundant code.